src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -824,9 +824,14 @@ private static int IndexOfNonWhiteSpaceCharacter(ITextSnapshotLine line)
                         var executionResult = await Evaluator.ExecuteCodeAsync(snapshotSpan.GetText()).ConfigureAwait(true);
                         Debug.Assert(_window.OnUIThread()); // ConfigureAwait should bring us back to the UI thread.
 
-                        // For reset command typed at prompt, the state should be WaitingForInput 
-                        // and for all other submissions it should be Executing input
-                        Debug.Assert(State == State.ExecutingInput || State == State.WaitingForInput, $"Unexpected state {State}");
+                        // For reset command typed at prompt -> the state should be WaitingForInput. 
+                        // For all other submissions on the prompt -> it should be Executing input.
+                        // If reset button is clicked during a long running submission -> it could be Resetting becasue 
```

Comment:
No action is required in the Resetting case because Reset will take us back to WaitingForInput, correct?